dataset/st_dataset.py

```diff
@@ -155,17 +155,17 @@ def get_dataset_info(data_dict: DatasetDict) -> DatasetInfo:
 
 
 
-def load_dataset_safe(*args, **kwargs) -> Dataset:
+def load_dataset_safe(*args, **kwargs, numtries=3) -> Dataset:
 
-    tries = 3
+    tries = numtries
     for i in range(tries):
         try:
             ds = load_dataset(*args, *kwargs)
         except :
             if i < tries - 1: # i is zero indexed
                 continue
             else:
-                sys.exit("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
+                raise RuntimeError("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
```

Comment:
@niansong1996 It's unclear how to utilize both 'raise' and 'return' in the same clause so I opted to use 'raise'. Let me know what you think

Reply:
Ah, yes. That's a typo from me, you can't do `raise` and `return` in the same block of code. This LGTM